docs/samples/Microsoft.ML.Samples/Dynamic/BinaryClassification/AveragedPerceptron.cs

```diff
@@ -0,0 +1,61 @@
+﻿using Microsoft.ML;
+
+namespace Microsoft.ML.Samples.Dynamic.BinaryClassification
+{
+    public static class AveragedPerceptron
+    {
+        public static void Example()
+        {
+            // In this examples we will use the adult income dataset. The goal is to predict
+            // if a person's income is above $50K or not, based on different pieces of information about that person.
+            // For more details about this dataset, please see https://archive.ics.uci.edu/ml/datasets/adult
+
+            // Create a new context for ML.NET operations. It can be used for exception tracking and logging, 
+            // as a catalog of available operations and as the source of randomness.
+            // Setting the seed to a fixed number in this examples to make outputs deterministic.
+            var mlContext = new MLContext(seed: 0);
+
+            // Download the dataset and load it as IDataView
+            var data = SamplesUtils.DatasetUtils.LoadAdultDataset(mlContext);
+
+            // Leave out 10% of data for testing
+            var (trainData, testData) = mlContext.BinaryClassification.TrainTestSplit(data, testFraction: 0.1);
+
+            // Create data processing pipeline
+            var pipeline =
+                // Convert categorical features to one-hot vectors
+                mlContext.Transforms.Categorical.OneHotEncoding("workclass")
+                .Append(mlContext.Transforms.Categorical.OneHotEncoding("education"))
+                .Append(mlContext.Transforms.Categorical.OneHotEncoding("marital-status"))
+                .Append(mlContext.Transforms.Categorical.OneHotEncoding("occupation"))
+                .Append(mlContext.Transforms.Categorical.OneHotEncoding("relationship"))
+                .Append(mlContext.Transforms.Categorical.OneHotEncoding("ethnicity"))
+                .Append(mlContext.Transforms.Categorical.OneHotEncoding("native-country"))
```

Comment:
Can `OneHotEncoding()` take a set of column to transform? It might be cleaner.

Reply:
I'd like that too. The current way to transform multiple columns in one transform is to use the overload that takes an array of ColumnInfo objects. It can't be done with just strings. I think this longer version is more readable for samples. Involving ColumnInfo is a more advanced usage and is not the main focus of this sample.

If you want to see the API that you like, please add an issue for project 13. 

---
In reply to: [255265144](https://github.com/dotnet/machinelearning/pull/2483#discussion_r255265144) [](ancestors = 255265144)

Reply:
If the ColumnInfo form of the API isn't better (or good enough for docs) we should revisit the topic. I certainly see and agree to your point that the longer form is more readable.  I expect the current longer form takes N-passes of the data, which is sub-optimal. Perhaps we could make a more readable multi-input API form?

@TomFinley -- thoughts?

Reply:
to be clear, I'm not saying ColumnInfo is not good for docs. I'm saying that usage is not best for this sample which is about a binary classification task. A usage of ColumnInfo will be part of samples for OneHotEncoding.

---
In reply to: [255756620](https://github.com/dotnet/machinelearning/pull/2483#discussion_r255756620) [](ancestors = 255756620)